.hooks/pre-commit

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+#
+CXX_FORMAT="clang-format"
+CXX_EXTS=".c .h .cpp .hpp .cc .hh .cxx .hxx"
+
+PY_FORMAT="autopep8"
+PY_EXTS=".py"
+
+command_exists() {
+	hash $1 2>/dev/null
```

Comment:
I was confused by this post: http://stackoverflow.com/questions/592620/check-if-a-program-exists-from-a-bash-script. Which one is the best?